pkg/frontend/querymiddleware/request_validation_test.go

```diff
@@ -209,9 +209,13 @@ func TestCardinalityQueryRequestValidationRoundTripper(t *testing.T) {
 		},
 		{
 			// non-utf8 label name will be rejected even when we transition to UTF-8 label names
-			url:             cardinalityLabelValuesPathSuffix + "?label_names[]=\\xbd\\xb2\\x3d\\xbc\\x20\\xe2\\x8c\\x98",
+			url:             cardinalityLabelValuesPathSuffix + "?label_names[]=\xbd\xb2\x3d\xbc\x20\xe2\x8c\x98",
```

Comment:
also can you make the smae changes to `TestMetricsQueryRequestValidationRoundTripper`  and `TestLabelsQueryRequestValidationRoundTripper`? maybe also in `TestPrometheusCodec_DecodeEncodeMultipleTimes_Labels`

Reply:
oh, i'm surprised we don't do any validation on this request? do we delegate it to prometheus?

Reply:
As far as I can see, the request is not validated in `querymiddleware`. It looks like prometheus' `web/api/v1.API` already does validation on the query name, for label values, label names and series endpoints.